tests/integration/api/server.spec.js

```diff
@@ -230,4 +233,176 @@ describe('server', () => {
       await utils.listenForApi();
     });
   });
+
+  describe('Request UUID propagated to audit layer', () => {
+    const getReqId = (logLine) => {
+      const re = /\b[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}\b/;
+      if (re.test(logLine)) {
+        const match = logLine.match(re);
+        return match?.[0];
+      }
+    };
+
+    describe('for online users', () => {
+      it('should propagate uuid via proxy', async () => {
+        const collectApiLogs = await utils.collectApiLogs(/\/_all_docs\?limit=1/);
+        const collectHaproxyLogs = await utils.collectHaproxyLogs(/\/_all_docs\?limit=1/);
+
+        const result = await utils.request('/medic/_all_docs?limit=1');
+        await utils.delayPromise(500); // wait for everything to get logged
```

Comment:
This will need to introduce some kind of passing what we expect and a timeout. 
It adds some complexity for half a second. Do you think it's justified?

Reply:
Definitely. It's a pattern we use a lot so if we nail it then we've added some complexity in a util somewhere, but save 0.5s in many places. It doesn't have to be part of this issue - feel free to raise a new one and replace this pattern everywhere at once.

Reply:
I've created the issue. https://github.com/medic/cht-core/issues/9620
I think this is quite an intricate and possibly e2e test destabilizing. I'd rather have this PR clean and merged.